applications/conceptual-model-editor/src/app/diagram/dialog/select-classes-dialog.tsx

```diff
@@ -0,0 +1,117 @@
+
+    const localOpen = (onSelectConfirmCallback?: (newEntityID: string[]) => void) => {
+        setOnSelectConfirmCallback(onSelectConfirmCallback);
+        open();
+    };
+
+    // TODO: Maybe could be optimized later, when I get to know React better
+    function createChangedRecord<T>(record: Record<string, T>, id: string, newValue: T): Record<string, T> {
+        const newRecord = {...record};
+        newRecord[id] = newValue;
+        return newRecord;
+    }
+
+
+    const SelectClassesDialog = () => {   
+        const [connectionsMap, setConnectionsMap] = useState<Record<string, boolean>>({});
+
+        const { models, aggregatorView } = useModelGraphContext();
+        const activeVisualModel = aggregatorView.getActiveVisualModel();     
+
+        const { language: preferredLanguage } = useConfigurationContext();
+
+        const getPreferredName = useCallback((entity: EntityDetailSupportedType) => {
+            const { name } = EntityProxy(entity, preferredLanguage);
+            return name;
+        }, [preferredLanguage]);
+            
+        
+        // TODO: Probably could use better dependecies, but if it works it works
+        const getSelectedClasses = useCallback(() => {
+            const selectedClasses = Object.entries(connectionsMap).map(([k, v]) => {                
+                if(v) {
+                    return k;
+                }
+            });
+            if(selectedClasses === undefined) {
+                return [];
+            }
+            else {
+                return selectedClasses.filter(e => e !== undefined);
+            }
+        }, [connectionsMap]);
+
+        const onAcceptCallback = useCallback(() => {
+            if(onSelectConfirmCallback === undefined) {                
+                close();                
+            }
+            else {
+                onSelectConfirmCallback(getSelectedClasses());
+                close();
+            }            
+        }, [getSelectedClasses]);
+           
+
+        // TODO: Probably will have to show profiles later differently (like in catalog - using TreeLikeOffset)
+        return (
+            <BaseDialog heading={t("select-classes-dialog.select-classes")}>
+                <div>
+                    {/* We could also modify and then reuse <EntityCatalog />, it would be kind of nice, but it would take a lot of time */}
+                    <ul>
+                        {[...models.entries()].map(([modelId, model]) => 
+                            <div key={`select-classes-dialog-div-${model.getId()}`}>
+                                <h4 key={`select-classes-dialog-h4-${model.getId()}`}>Ⓜ {model.getAlias() ?? ""}</h4>
+                                {Object.values(model.getEntities()).map(e => {
+                                    if(isSemanticModelClass(e) || isSemanticModelClassUsage(e)) {
```

Comment:
I would extract this into a method .. all together it seems a little complicated to me.